src/WrappedMToken.sol

```diff
@@ -700,14 +701,38 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
     function _getSafeTransferableM(uint240 amount_, uint128 currentIndex_) internal view returns (uint240 safeAmount_) {
         // If this contract is earning, adjust `amount_` to ensure it's M balance decrement is limited to `amount_`.
         return
-            IMTokenLike(mToken).isEarning(address(this))
+            _isEarningM(address(this))
                 ? IndexingMath.getPresentAmountRoundedDown(
                     IndexingMath.getPrincipalAmountRoundedDown(amount_, currentIndex_),
                     currentIndex_
                 )
                 : amount_;
     }
 
+    /**
+     * @dev    Compute the adjusted amount of M that must be transferred so the recipient receives at least that amount.
+     * @param  recipient_        The address of some recipient.
+     * @param  amount_           Some amount to be transferred out of the wrapper.
+     * @return sufficientAmount_ The adjusted amount that must be transferred.
+     */
+    function _getSufficientTransferableM(
+        address recipient_,
+        uint240 amount_
+    ) internal view returns (uint240 sufficientAmount_) {
+        uint128 currentMIndex_ = _currentMIndex();
+        uint112 principal = uint112(IMTokenLike(mToken).principalBalanceOf(recipient_));
+        uint240 balance = IndexingMath.getPresentAmountRoundedDown(principal, currentMIndex_);
+
+        // If the recipient is earning, adjust `amount_` to ensure it's M balance increments by at least `amount_`.
+        return
```

Comment:
unchecked? 
but more important to discuss the necessity of this method in general

Reply:
It is important as a safety to ensure consistent received amount behaviour in M if and when earning is disaabled. The latest version of this OR reduces reliance on this fucntion in other cases.